### PR TITLE
[Module]: Azure Database for PostgreSQL Flexible Servers Module

### LINKED ISSUE #1751: [Feature Request]: Develop Microsoft Database for PostgreSQL Module
### Description

We do not have a module for this yet in the library.

It contains two types:
- Flexible
- Servers

https://docs.microsoft.com/en-us/azure/templates/microsoft.dbforpostgresql

### PR DESCRIPTION
**Pull request by ahmadabdalla**
# Description

Closes #1751 

- Enabled the PostgreSQL - Flexible Server Module 

- Added a new dependency vnet and a private dns zone for the module.
- Contains 3 child modules (configurations, firewallrules, databases)
- Contains 3 scenarios for parameter file testing to cover all parameter combinations

**NOTE**: The nature of the of this resource type 'may' sometimes fail when adding the 'configurations', especially when having more than one in the array. This is not a CARML issue. The error returned is 'InternalServerError'. However, after performing multiple tests. The likelihood of this happening is small. I suggested initially whether we should remove this parameter from being tested, but I prefer to have it. Our pipelines retry 3 times for a deployment and that seems to be sufficient

| Pipeline |
| - |
|[DbForPostgreSQL: FlexibleServers](https://github.com/ahmadabdalla/ResourceModules/actions/workflows/ms.dbforpostgresql.flexibleservers.yml) |

image

# Type of Change

- [ ] New feature (non-breaking change which adds functionality)

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/Microsoft.DBforPostgreSQL/flexibleServers/.bicep/nested_roleAssignments.bicep [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+resource flexibleServer 'Microsoft.DBforPostgreSQL/flexibleServers@2022-01-20-preview' existing = {
+  name: last(split(resourceId, '/'))
+}
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = [for principalId in principalIds: {
+  name: guid(flexibleServer.id, principalId, roleDefinitionIdOrName)
+  properties: {
+    description: description
```
Comment (MrMCake) [inline modules/Microsoft.DBforPostgreSQL/flexibleServers/.bicep/nested_roleAssignments.bicep:47]: Could you align this with the latest version in the other modules? With the recent upgrade of the Api version we introduced all missing properties

File: modules/Microsoft.DBforPostgreSQL/flexibleServers/configurations/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,51 @@
+@description('Required. The name of the database.')
```
Comment (MrMCake) [inline modules/Microsoft.DBforPostgreSQL/flexibleServers/configurations/deploy.bicep:1]: Rather the name of the configuration,  I guess

File: modules/Microsoft.DBforPostgreSQL/flexibleServers/.bicep/nested_roleAssignments.bicep (hunk 1 above)
Comment (ahmadabdalla) [reply modules/Microsoft.DBforPostgreSQL/flexibleServers/.bicep/nested_roleAssignments.bicep:59]: Done

File: modules/Microsoft.DBforPostgreSQL/flexibleServers/configurations/deploy.bicep (hunk 2 above)
Comment (ahmadabdalla) [reply modules/Microsoft.DBforPostgreSQL/flexibleServers/configurations/deploy.bicep]: Good pickup. Fixed

**Review by MrMCake — APPROVED** [review]

**Pull request merged**